packages/story-editor/src/components/colorPicker/basicColorList.js

```diff
@@ -44,26 +44,66 @@ const focusStyle = css`
     )};
 `;
 
-const SwatchList = styled.div.attrs({
-  role: 'listbox',
```

Comment:
> This is fine if there are other options in the listbox, but when there are no options (like in the e2e test for saved colors) then it fails the axe test because the listbox has no children with the role group or option.

I think I asked it somewhere already but can't find exactly where / what was the answer -- couldn't we just change the role of the Add button to `option` and leave everything else as it was?

Alternatively, could we display the `listbox` + the Add button inside it only if there are colors, and display just the one button without a listbox if there aren't any?

Reply:
> > This is fine if there are other options in the listbox, but when there are no options (like in the e2e test for saved colors) then it fails the axe test because the listbox has no children with the role group or option.
> 
> I think I asked it somewhere already but can't find exactly where / what was the answer -- couldn't we just change the role of the Add button to `option` and leave everything else as it was?

For the `Add Color` button to be an `option` it would have to have an `aria-selected` property too - which it can't because it's not selectable. I think making it an option wouldn't make sense to screen reading technologies.

> Alternatively, could we display the `listbox` + the Add button inside it only if there are colors, and display just the one button without a listbox if there aren't any? Not sure if this would be ok from accessibility point of view (cc @barklund)

Hmm good idea - I'll investigate this

Reply:
> For the Add Color button to be an option it would have to have an aria-selected property too - which it can't because it's not selectable. I think making it an option wouldn't make sense to screen reading technologies.

Makes sense!

Reply:
> Alternatively, could we display the `listbox` + the Add button inside it only if there are colors, and display just the one button without a listbox if there aren't any? 

This passes axe tests and is much simpler 🥳

Reply:
Thanks for putting all this effort into fixing something I broke 😛 

I added a separate issue for myself to fix the navigation with arrow keys: https://github.com/google/web-stories-wp/issues/9701